app/src/main/java/com/lagradost/cloudstream3/utils/VideoDownloadManager.kt

```diff
@@ -1705,7 +1710,28 @@ object VideoDownloadManager {
         }
     }
 
-    fun deleteFileAndUpdateSettings(context: Context, id: Int): Boolean {
+    fun deleteFilesAndUpdateSettings(context: Context, ids: List<Int>, scope: CoroutineScope) {
+        scope.launchSafe(Dispatchers.IO) {
+            val deleteJobs = ids.map { id ->
```

Comment:
This just looks like you reinvented the amap function.

Reply:
Sorta I suppose, the difference was from my understanding at least, is amap defined here, uses a GlobalScope, here, we allow passing a CoroutineScope  which we use to pass a ViewModelScope, which since the view model handles the deletion here it provided a bit better performance from my observations. We also use it to pass a MainScope in single deletions which seemed to be better performance for single deletions as well then the old method directly.

I could be wrong in my understanding though.

Reply:
@fire-light42 is this okay or do you still want me to do something here by the way?